extensions/statistics/src/Api/Controller/ShowStatisticsData.php

```diff
@@ -69,20 +70,36 @@ public function handle(ServerRequestInterface $request): ResponseInterface
         // control panel.
         $actor->assertAdmin();
 
-        $reportingPeriod = Arr::get($request->getQueryParams(), 'period');
-        $model = Arr::get($request->getQueryParams(), 'model');
+        $query = $request->getQueryParams();
 
-        return new JsonResponse($this->getResponse($model, $reportingPeriod));
+        $reportingPeriod = Arr::get($query, 'period');
+        $model = Arr::get($query, 'model');
+        $customDateRange = Arr::get($query, 'dateRange');
+
+        return new JsonResponse($this->getResponse($model, $reportingPeriod, $customDateRange));
     }
 
-    private function getResponse(?string $model, ?string $period): array
+    private function getResponse(?string $model, ?string $period, ?array $customDateRange): array
     {
         if ($period === 'lifetime') {
             return $this->getLifetimeStatistics();
         }
 
         if (! Arr::exists($this->entities, $model)) {
-            throw new InvalidParameterException();
+            throw new InvalidParameterException('A model must be specified');
+        }
+
+        if ($period === 'custom') {
+            if (! $customDateRange) {
+                throw new InvalidParameterException('A custom date range must be specified');
+            }
+
+            // Seconds-based timestamps
+            $startRange = Carbon::createFromTimestampUTC($customDateRange['start'])->toDateTime();
+            $endRange = Carbon::createFromTimestampUTC($customDateRange['end'])->toDateTime();
```

Comment:
Just to make sure use input is exactly what we expect it to be
```suggestion
        if ($period === 'custom') {
            $start = (int) $customDateRange['start'];
            $end = (int) $customDateRange['end'];

            if (! $customDateRange || ! $start || ! $end) {
                throw new InvalidParameterException('A custom date range must be specified');
            }

            // Seconds-based timestamps
            $startRange = Carbon::createFromTimestampUTC($start)->toDateTime();
            $endRange = Carbon::createFromTimestampUTC($end)->toDateTime();
```